src/Symfony/Bundle/WebServerBundle/WebServerConfig.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebServerBundle;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class WebServerConfig
+{
+    private $hostname;
+    private $port;
+    private $documentRoot;
+    private $env;
+    private $router;
+
+    public function __construct($documentRoot, $env, $address = null, $router = null)
+    {
+        if (!is_dir($documentRoot)) {
+            throw new \InvalidArgumentException(sprintf('The document root directory "%s" does not exist.', $documentRoot));
+        }
+
+        if (null === $file = $this->guessFrontController($documentRoot, $env)) {
+            throw new \InvalidArgumentException(sprintf('Unable to guess the front controller under "%s".', $documentRoot));
+        }
+
+        putenv('APP_FRONT_CONTROLLER='.$file);
+
+        $this->documentRoot = $documentRoot;
+        $this->env = $env;
+        $this->router = $router ?: __DIR__.'/Resources/router.php';
+
+        if (null === $address) {
+            $this->address = '127.0.0.1';
+            $this->port = $this->findBestPort();
+            $address = $this->address.':'.$this->port;
```

Comment:
you should return here, or put this ``if`` in the same chain than others, to avoid parsing the address and port again